### ISSUE TITLE
Output shape mismatch for model trained with DNNEstimator

### ISSUE THREAD

**Issue by shawpan** [tensorflow/serving#1499]
### System information
- **macOS Sierra 10.12.6, Amazon Linux AMI 2018.03**:
- **TensorFlow Serving installed from : binary**:
- **TensorFlow Serving version 1.14.0**:

### Describe the problem
tf-serving throws output shape mismatch error for `DNNLinearCombinedEstimator` model with 
multi label regression head.

> { "error": "Expected output Tensor shape to be either [batch_size] or [batch_size, 1] but got [1,2]" }% 

### Exact Steps to Reproduce
1. train a model using DNNLinearCombinedEstimator with custom `tf.contrib.estimator.regression_head`  and `label_dimension=2`.
2. Serve the model using tf-serving from docker
3. Request the `regress` api 
4. It will throw error because of >1 output labels

> { "error": "Expected output Tensor shape to be either [batch_size] or [batch_size, 1] but got [1,2]" }% 

### Source code / logs

`https://github.com/shawpan/test_tf`
 has all the codes to reproduce this issue. 
`https://github.com/shawpan/test_tf/blob/master/test_DNNLinearCombinedEstimator.py` 
contains the model building, training and exporting scripts.
`https://github.com/shawpan/test_tf/tree/master/export/1` has the saved model.
`https://github.com/shawpan/test_tf/blob/master/deploy.sh` deploys the model on local `tf-serving` docker 
`https://github.com/shawpan/test_tf/blob/master/do_request.sh` do a curl request on regress api.

**Comment by gowthamkpr** [comment]
@shawpan Can you please give us a simple example to reproduce your issue and also can you follow the template [here](https://github.com/tensorflow/serving/issues/new) while creating the issue as this is too vague for us to help you in this issue. Thanks!

**Comment by shawpan** [comment]
@gowthamkpr description updated and also the reproducible codes are given as a separate repository.

**Comment by shawpan** [comment]
@gowthamkpr did you check the reproducible example?
